README.md

```diff
@@ -92,14 +92,38 @@ an OAuth2 token (with the "gist" permission).
 
 This token is stored in `~/.gist` and used for all future gisting. If you need to
 you can revoke it from https://github.com/settings/tokens, or just delete the
-file.  If you need to store tokens for both github.com and a Github Enterprise instance 
-you can save your Github Enterprise token in `~/.gist.github.example.com` where 
-"github.example.com" is the URL for your Github Enterprise instance.
+file. 
 
 ‌After you've done this, you can still upload gists anonymously with `-a`.
 
     gist -a a.rb
 
+### GitHub Enterprise
+
+If you'd like `gist` to use your locally installed [GitHub Enterprise](https://enterprise.github.com/),
+you need to export the `GITHUB_URL` environment variable (usually done in your `~/.bashrc`).
+
+    export GITHUB_URL=http://github.internal.example.com/
+
+Once you've done this and restarted your terminal (or run `source ~/.bashrc`), gist will
+automatically use github enterprise instead of the public github.com
+
+Your token for GitHub Enterprise will be stored in `.gist.<protocol>.<server.name>[.<port>]` (e.g.
+`~.gist.http.github.internal.example.com` for the GITHUB_URL example above) instead of `~/.gist`.
+
+If you have multiple servers or use Enterprise and public GitHub often, you can work around this by creating scripts
+that set the env var and then run `gist`. Keep in mind that to use the public GitHub you must unset the env var. Just
+setting it to the public URL will not work. Use `unset GITHUB_URL`
+
+### Token file format
+
+If you cannot use passwords, as most Enterprise instalations do, you can generate the token via the web interface
+and then simply save the string in the correct file. Avoid line breaks or you migth see:
```

Comment:
not nit-pick at all. Thanks for spotting. Didn't have spell check enabled for md files...